src/testshade/testshade.cpp

```diff
@@ -390,15 +390,7 @@ action_groupspec (int argc, const char *argv[])
     if (OIIO::Filesystem::exists (groupspec)) {
         // If it names a file, use the contents of the file as the group
         // specification.
-        std::ifstream in;
-        OIIO::Filesystem::open (in, groupspec);
-        groupspec.clear();
-        while (in.good()) {
-            std::string line;
-            std::getline (in, line);
-            groupspec += line + "\n";
-        }
-        in.close ();
+        OIIO::Filesystem::read_text_file (groupspec, groupspec);
```

Comment:
It doesn't modify the string until after it's done with the filename.

The contents of Filesystem::read_text_file work precisely like the lines I removed, except for using rdbuf rather than getline, which means the new way is faster.